### PR TITLE
Teammate finder backend

### PR DESCRIPTION
**Pull request by techfreek**
Merge the node.js app that handles the teammate finding system. Does not include the front end updates.

### CODE REVIEW

**Comment by techfreek** [comment]
@whyrusleeping or @dylanPowers can we merge this?

File: server.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+			description = null;
+			captain = null;
+			
+			if(team.creator.emails[0].email === user.profile.email) {
+				captain = {
+					name: user.profile.name,
+					email: user.profile.email
+				};
+
+				user.answers.forEach(function(question) {
```
Comment (dylanPowers) [inline server.js]: You've got like 3 nested loops here. Pulling these out into functions so there's not so much nesting would increase readability.

**Comment by dylanPowers** [comment]
Tabs vs spaces...we need to pick one. I'm use to Dart where the convention is 2 spaces and I've used that already in the Gulpfile.

**Comment by techfreek** [comment]
I'll take care of the loops this afternoon. And sorry, I am used to tabs but I'll switch to two spaces.

**Pull request merged**